### ISSUE TITLE
LoginProducer.java:38-40: Make sure to extract the...

### ISSUE THREAD

**Issue by 0pdd** [self-xdsd/self-web#28]
The puzzle `25-55626d2f` from #25 has to be resolved: 

https://github.com/self-xdsd/self-web/blob/0085e22d3a222b9a2acb0906cd948b3677f3d2c0/src/main/java/com/selfxdsd/selfweb/LoginProducer.java#L38-L40

The puzzle was created by @amihaiemil on 22-Jul-20. 

Estimate: 30 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

**Comment by 0pdd** [comment]
@amihaiemil please pay attention to this new issue.

**Comment by criske** [comment]
@amihaiemil I'll work on this and also implement the GitlabLogin plus the application.properties updated for gitlab. 

I've played a bit with the app, but I couldn't figure out where to inject the `Login` bean (worked very little with Spring in the past). So I've changed a little bit the design just to check if the login works for my Github and Gitlab accounts.
[LoginProducer](https://gist.github.com/criske/a035cc616e23cc96fb416f9f8f1b9d19)
Token is received via an `ApplicationListener` event.
[AuthenticationSuccessListener](https://gist.github.com/criske/27e99a2c51614968bd01f52da52157e2)

The PR though will only refactor the code the as the puzzle says and will update the `application.properties` with gitlab client settings commented. :)

**Comment by amihaiemil** [comment]
@criske wait a little, don't start. 

I make some changes to how the Login is used -- we will inject the User already logged in, instead of the Login and the core separately. This way, login will happen only once per session.

**Comment by amihaiemil** [comment]
@criske Ok, you can see now how LoginProducer gives us the authenticated User directly. See how it's used in Repositories class, for instance.

**Comment by amihaiemil** [comment]
@criske Don't bother with the GitLab Oauth flow too much. Just the puzzle's description is enough :D

**Comment by criske** [comment]
@amihaiemil looks much better now, so `Self` is now "lazly" logged-in. 
And about GitLab Oauth flow, I did it for GitlabLogin, especially for the user `username` field from Principal, which is different than the Github one. I'll keep the Gitlab config around, just let me know when we needed it.

**Comment by 0pdd** [comment]
The puzzle `25-55626d2f` has disappeared from the source code, that's why I closed this issue. //cc @amihaiemil

**Comment by amihaiemil** [comment]
Paid with transaction id 2C5721509J103993S
